File: Exercices/Threshold/Python/threshold.py
#!/usr/bin/env python

import sys
import SimpleITK

if len(sys.argv) != 4:
  print sys.argv[0]+" <inputImage> <outputImage> <thresholdValue>"
  sys.exit(1)
# Input Image
reader = SimpleITK.ImageFileReader(); 
reader.SetFileName(sys.argv[1]); 
image = reader.Execute();

# Threshold Filter 
thresholdFilter = SimpleITK.BinaryThresholdImageFilter();
thresholdFilter.SetLowerThreshold(int(sys.argv[3])); 
thresholdFilter.SetUpperThreshold(255); 
thresholdFilter.SetOutsideValue(0); 
thresholdFilter.SetInsideValue(255); 

# Output Image
output = thresholdFilter.Execute(image);
writer = SimpleITK.ImageFileWriter(); 
writer.SetFileName(sys.argv[2]); 
writer.Execute(output); 

File: Exercices/Threshold/C++/Threshold.cxx
#include "itkImage.h"
#include "itkBinaryThresholdImageFilter.h"
#include "itkImageFileReader.h"
#include "itkImageFileWriter.h"
#include "ThresholdCLP.h"

int main(int argc, char *argv[])
{	
	PARSE_ARGS;	

	//Seuil//
	int Threshold=atoi(argv[3]);
 
	//Image Type//
	typedef itk::Image<unsigned char, 3> ImageType;

	//Input Image//
	typedef itk::ImageFileReader<ImageType> ReaderType; 
	ReaderType::Pointer reader = ReaderType::New(); 
	reader->SetFileName(argv[1]); 

	//Threshold Filtering//
	typedef itk::BinaryThresholdImageFilter <ImageType, ImageType> FilterType;
	FilterType::Pointer thresholdFilter=FilterType::New(); 
	
	thresholdFilter->SetInput(reader->GetOutput());

	thresholdFilter->SetLowerThreshold(Threshold);
	thresholdFilter->SetUpperThreshold(255);

	thresholdFilter->SetOutsideValue(0); 
	thresholdFilter->SetInsideValue(255); 
	
	//Output Image
	typedef itk::ImageFileWriter<ImageType> WriterType;  
	WriterType::Pointer writer = WriterType::New(); 
	writer->SetInput(thresholdFilter->GetOutput()); 
	writer->SetFileName(argv[2]);
	writer->Update(); 
}
